gcloud-java-storage/src/test/java/com/google/gcloud/storage/RemoteGcsHelper.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.storage;
+
+import com.google.gcloud.AuthCredentials;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Utility to create a remote storage configuration for testing
+ */
+public class RemoteGcsHelper {
+
+  private static final Logger log = Logger.getLogger(RemoteGcsHelper.class.getName());
+  private static final String BUCKET_NAME_PREFIX = "gcloud-test-bucket-temp-";
+  private static final String PROJECT_ID_ENV_VAR = "GCLOUD_TESTS_PROJECT_ID";
+  private static final String PRIVATE_KEY_ENV_VAR = "GCLOUD_TESTS_KEY";
+
+  private final StorageOptions options;
+  private final String bucket;
+
+  private RemoteGcsHelper(StorageOptions options, String bucket) {
+    this.options = options;
+    this.bucket = bucket;
+  }
+
+  public StorageOptions options() {
+    return options;
+  }
+
+  public String bucket() {
+    return bucket;
+  }
+  
+  private static String generateBucketName() {
```

Comment:
I see your point but I do not agree completely. I do not expect people other than us and possibly some contributors to write integration tests. If such tests are meant to be pushed to the repository it's undesirable that bucket name is hardcoded in the test code.